lib/spack/spack/util/compression.py

```diff
@@ -234,7 +246,15 @@ def _7zip(archive_file):
     Args:
         archive_file (str): absolute path of file to be unarchived
     """
+    # 7zip will decompress and extract .txz atomically
+    # but not .tar.xz, rename tar.xz
+    # to .txz for atomic decompression
+    if "tar.xz" in archive_file:
+        file_path = archive_file.strip(".tar.xz")
+        os.rename(archive_file, file_path + ".txz")
```

Comment:
A couple points:

* This has the same problem as https://github.com/spack/spack/pull/30690#discussion_r899382068. For now, the old file should be copied (or symlinked).
* `strip()` is a generally fragile when used for this: note that this looks like it would strip instances of the string ".tar.xz", but in fact ".tar.xz" is treated as a set of characters to strip - any order/combination of characters in that set will be removed

In all I recommend:

```
if ".tar.xz" in archive_file: #note the extra period at the beginning
  new_path = os.path.join(tmpdir, os.path.basename(archive_file.replace(".tar.gz", ".txz")))
  os.copy(archive_file, new_path)
  archive_file = new_path
```

Reply:
I think it makes more sense to fold this change into #31589 and rely on that PR prior to merging this one. The issue resolved by this (arguably hacky) workaround is the same as is being addressed in more recent changes in that PR.